bzzeth/bzzeth_test.go

```diff
@@ -121,37 +130,39 @@ func TestBzzEthHandshake(t *testing.T) {
 	}
 }
 
-// TestBzzBzzHandshake tests that a handshake between two Swarm nodes
-func TestBzzBzzHandshake(t *testing.T) {
+// TestBzzBzzHandshakeWithMessage tests that a handshake between two Swarm nodes and message exchange
+// disconnects the peer
+func TestBzzBzzHandshakeWithMessage(t *testing.T) {
+	// redefine isSwarmNodeFunc to force recognise remote peer as swarm node
+	isSwarmNodeFunc = func(_ *Peer) bool { return true }
```

Comment:
Should isSwarmNodeFunc be reset to the original value after the test is done?

Reply:
Can be reset. But i don't see the need as we create a new BZZ service for every testcase.

Reply:
I see that you removed the reset in this PR. Regardless of the purpose, if something global is changed it is polite to revert it back when the test is finished with it. What would happen if another test is made that relies implicitly on default value of isSwarmNodeFunc? The order of test functions would matter as no test function could rely what is the value of isSwarmNodeFunc.

Reply:
Done